Sources/Subs-Db-postgresql.php

```diff
@@ -717,6 +717,11 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 	// Replace the prefix holder with the actual prefix.
 	$table = str_replace('{db_prefix}', $db_prefix, $table);
 
+	// Sanity check for replace is key part of the columns array
+	if ($method == 'replace' && count(array_intersect_key($columns, array_flip($keys))) !== count($keys))
+		smf_db_error_backtrace('Primary Key field missing in insert call',
+				'Change the methode of db insert to insert or add the pk field to the columns array', E_USER_ERROR, __FILE__, __LINE__);
```

Comment:
These strings should be loaded using the `$txt` variable instead of hard-coding the English text.

Reply:
When you look in the other backtrace you notice that every where is written in hardcode and
specialy database stuff is not the rightplace for this.

When you wanted feel free to do this but than please for the complet backtrace stuff.

Reply:
Fair enough. Then please just change "methode" to "method".